Extensions/read_posts.js

```diff
@@ -20,51 +20,48 @@ XKit.extensions.read_posts = new Object({
 
 	dashboard_regex: new RegExp("^https?://www.tumblr.com/dashboard(?:$|/)\\#?"),
 
-	last_undimmed: "",
+	undimmed_post: null,
+	currently_undimming: false,
 
 	run: function() {
 		XKit.tools.init_css('read_posts');
 		XKit.post_listener.add('read_posts_process', this.process_posts);
 		this.process_posts();
- 		$(document).on("click",".post_notes_inner *, .post_control.reply, .tumblelog_menu_button, .share_social_button, .post_control_menu", XKit.extensions.read_posts.undim);
- 		// $(document).on("click",".popover_inner *, #glass_overlay", XKit.extensions.read_posts.redim);
+		$(document).on("click",".post_notes_inner *, .post_control.reply, .tumblelog_menu_button, .share_social_button, .post_control_menu", XKit.extensions.read_posts.undim);
+		$(document).on("click", XKit.extensions.read_posts.redim);
 
 		this.running = true;
 	},
 
 	redim: function(e) {
-
-		// This is broken and meh.
-		var m_obj = $(XKit.extensions.read_posts.last_dimmed).find(".post");
-		if (!$(XKit.extensions.read_posts.last_dimmed).hasClass("post")) {
-			var m_obj = $(m_obj).parentsUntil('.post').parent();
-		}
-		if (!$(m_obj).hasClass("post")) {
-			// STILL?!
-			return;
+		// This is less broken and meh.
+		if (XKit.extensions.read_posts.undimmed_post == null || XKit.extensions.read_posts.currently_undimming) {
```

Comment:
The `==` here could be marked as a jshint error (I'm also trying to review #155), so we could use `!XKit.extensions.read_posts.undimmed_post` instead.